sdk/python/foundation-models/system/inference/zero-shot-image-classification/zero-shot-image-classification-online-endpoint.ipynb

```diff
@@ -0,0 +1,357 @@
+    "# Create an online endpoint\n",
+    "endpoint = ManagedOnlineEndpoint(\n",
+    "    name=online_endpoint_name,\n",
+    "    description=\"Online endpoint for \"\n",
+    "    + foundation_model.name\n",
+    "    + \", for zero-shot-image-classification task\",\n",
+    "    auth_mode=\"key\",\n",
+    ")\n",
+    "workspace_ml_client.begin_create_or_update(endpoint).wait()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azure.ai.ml.entities import OnlineRequestSettings, ProbeSettings\n",
+    "\n",
+    "deployment_name = \"zero-shot-image-mlflow-deploy\"\n",
+    "\n",
+    "# Create a deployment\n",
+    "demo_deployment = ManagedOnlineDeployment(\n",
+    "    name=deployment_name,\n",
+    "    endpoint_name=online_endpoint_name,\n",
+    "    model=foundation_model.id,\n",
+    "    instance_type=\"Standard_DS3_V2\",  # Use GPU instance type like Standard_NC6s_v3 for faster inference\n",
+    "    instance_count=1,\n",
+    "    request_settings=OnlineRequestSettings(\n",
+    "        max_concurrent_requests_per_instance=1,\n",
+    "        request_timeout_ms=90000,\n",
+    "        max_queue_wait_ms=500,\n",
+    "    ),\n",
+    "    liveness_probe=ProbeSettings(\n",
+    "        failure_threshold=49,\n",
+    "        success_threshold=1,\n",
+    "        timeout=299,\n",
+    "        period=180,\n",
+    "        initial_delay=180,\n",
+    "    ),\n",
+    "    readiness_probe=ProbeSettings(\n",
+    "        failure_threshold=10,\n",
+    "        success_threshold=1,\n",
+    "        timeout=10,\n",
+    "        period=10,\n",
+    "        initial_delay=10,\n",
+    "    ),\n",
+    ")\n",
+    "workspace_ml_client.online_deployments.begin_create_or_update(demo_deployment).wait()\n",
+    "endpoint.traffic = {deployment_name: 100}\n",
+    "workspace_ml_client.begin_create_or_update(endpoint).result()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 5. Test the endpoint\n",
+    "\n",
+    "We will fetch some sample data from the test dataset and submit to online endpoint for inference."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import base64\n",
+    "import json\n",
+    "\n",
+    "sample_image = os.path.join(dataset_dir, \"milk_bottle\", \"99.jpg\")\n",
+    "labels = \"water_bottle,milk_bottle,carton,can\"\n",
+    "\n",
+    "\n",
+    "def read_image(image_path):\n",
+    "    with open(image_path, \"rb\") as f:\n",
+    "        return f.read()\n",
+    "\n",
+    "\n",
+    "request_json = {\n",
+    "    \"input_data\": {\n",
+    "        \"columns\": [\"image\", \"text\"],\n",
+    "        \"index\": [0],\n",
+    "        \"data\": [\n",
+    "            [base64.encodebytes(read_image(sample_image)).decode(\"utf-8\"), labels]\n",
```

Comment:
Should we add an example for invoking endpoint on multiple images at once ?

Reply:
I think it is a good idea. I added a second image and comments to explain the request format

Reply:
I excluded the text column in the model card. Should we keep it consistent in the notebooks and examples: https://ml.azure.com/registries/azmlft-dev-registry01/models/openai-clip-vit-base-patch32/version/3?tid=72f988bf-86f1-41af-91ab-2d7cd011db47#overview

Reply:
Follow up...turns out this doesn't fail!